### ISSUE TITLE
**BROKEN** connectd: gossip_store: bad checksum at offset 218892226:

### ISSUE THREAD

**Issue by grubles** [ElementsProject/lightning#5565]
CLN v0.12 release
Rocky Linux 9 
ppc64le

Plugins: summary, CLBOSS, peerswap

CLN crashed in the middle of the night so I don't have any steps to reproduce unfortunately. 

stdout log:
```
0x1001a657 send_backtrace
        common/daemon.c:33
0x10025e7f status_failed
        common/status.c:221
0x1001e43f gossip_store_next
        common/gossip_store.c:156
0x1000f9d3 maybe_from_gossip_store
        connectd/multiplex.c:530
0x1000fdff write_to_peer
        connectd/multiplex.c:966
0x100d322b next_plan
        ccan/ccan/io/io.c:59
0x100d3a1b io_do_always
        ccan/ccan/io/io.c:435
0x100d51ef handle_always
        ccan/ccan/io/poll.c:304
0x100d5853 io_loop
        ccan/ccan/io/poll.c:385
0x1000bfe7 main
        connectd/connectd.c:2063
0x7fff9e777ca3 ???
        ???:0
0x7fff9e777e8f ???
        ???:0
0xffffffffffffffff ???
        ???:
Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.terminate called after throwing an instance of 'std::runtime_error'
```

CLN log file backtrace:

```
02022-09-01T10:40:19.304Z **BROKEN** lightningd: FATAL SIGNAL 11 (version v0.12.0)
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: common/daemon.c:38 (send_backtrace) 0x100885df
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: common/daemon.c:46 (crashdump) 0x10088657
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x7fffa2200463
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x7fffa1c6c210
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: ccan/ccan/json_out/json_out.c:297 (json_out_addstr) 0x1016fca3
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: common/json_stream.c:186 (json_add_string) 0x10096217
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/notification.c:114 (warning_notification_serialize) 0x1002f5d7
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/notification.c:128 (notify_warning) 0x100309b3
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/log.c:443 (logv) 0x1002e14f
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/log.c:494 (log_) 0x1002deef
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/log_status.c:28 (log_status_msg) 0x1002f1f7
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/subd.c:513 (sd_msg_read) 0x1005cfab
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x1016a7cf
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x1016b0b3
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x1016b1bb
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x1016dce7
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/io_loop_with_timers.c:22 (io_loop_with_timers) 0x10025ffb
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: lightningd/lightningd.c:1195 (main) 0x1002bbbb
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x7fffa1bb7ca3
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0x7fffa1bb7e8f
2022-09-01T10:40:19.305Z **BROKEN** lightningd: backtrace: (null):0 ((null)) 0xffffffffffffffff
```

**Comment by vincenzopalazzo** [comment]
Mh! I think this error is caused by clboss this is a c++ API `std::runtime_error`, there is any logs regarding clboss?

**Comment by grubles** [comment]
Not that I can find. :-/ Nothing obvious in the log file and the stdout log exceeds my tmux scrollback so I can't inspect it beyond what I posted.

**Comment by vincenzopalazzo** [comment]
Maybe can be useful for you to open a issue in clboss at the same time, maybe @ZmnSCPxj know what is going on

**Comment by rustyrussell** [comment]
See also: #5572

**Comment by rustyrussell** [comment]
If this happens again, *please* save the gossip_store file *before* restart....

**Comment by endothermicdev** [comment]
This crash should be fixed with 4167fe8
